src/variant_detection/variant_detection.cpp

```diff
@@ -4,28 +4,92 @@
 #include "variant_detection/bam_functions.hpp"                      // for hasFlag* functions
 
 using seqan3::operator""_tag;
 
+void detect_junctions_in_short_reads_sam_file(std::vector<Junction> & junctions,
+                                              std::filesystem::path const & alignment_short_reads_file_path,
+                                              std::vector<detection_methods> const & methods,
+                                              uint64_t const min_var_length)
+{
+    // Open input alignment file
+    using my_fields = seqan3::fields<seqan3::field::flag,       // 2: FLAG
+                                     seqan3::field::ref_id,     // 3: RNAME
+                                     seqan3::field::ref_offset, // 4: POS
+                                     seqan3::field::mapq,       // 5: MAPQ
+                                     seqan3::field::header_ptr>;
+
+    seqan3::sam_file_input alignment_short_reads_file{alignment_short_reads_file_path, my_fields{}};
+
+    // Check that the file is sorted before proceeding.
+    if (alignment_short_reads_file.header().sorting != "coordinate")
+    {
+        throw seqan3::format_error{"ERROR: Input file must be sorted by coordinate (e.g. samtools sort)"};
+    }
+    uint16_t num_good = 0;
+
+    for (auto & record : alignment_short_reads_file)
+    {
+        seqan3::sam_flag const flag         = record.flag();                            // 2: FLAG
+        int32_t const ref_id                = record.reference_id().value_or(-1);       // 3: RNAME
+        int32_t const ref_pos               = record.reference_position().value_or(-1); // 4: POS
+        uint8_t const mapq                  = record.mapping_quality();                 // 5: MAPQ
+        auto const header_ptr               = record.header_ptr();
+        auto const ref_ids = header_ptr->ref_ids();
+
+        if (hasFlagUnmapped(flag) || hasFlagSecondary(flag) || hasFlagDuplicate(flag) || mapq < 20 ||
+            ref_id < 0 || ref_pos < 0)
+            continue;
+
+        std::string const ref_name = ref_ids[ref_id];
+        for (detection_methods method : methods) {
+            switch (method)
+            {
+                case detection_methods::cigar_string: // Detect junctions from CIGAR string
+                    seqan3::debug_stream << "The split cigar string method for short reads is not yet implemented.\n";
+                    break;
+                case detection_methods::split_read:     // Detect junctions from split read evidence (SA tag,
+                    seqan3::debug_stream << "The split read method for short reads is not yet implemented.\n";
+                    break;
+                case detection_methods::read_pairs: // Detect junctions from read pair evidence
+                    if (hasFlagProperMappedPairedReads(flag))
+                    {
+                        analyze_read_pair();
+                    }
```

Comment:
The `0x2` (properly mapped pair) flag is not well defined by the SAM specification but it usually means that the two mates are mapped in the right orientation and distance to each other. Analyzing only these reads, however, would be very boring and would ignore all the interesting reads covering SVs (which is what we actually want to find). So I would rather look at flag `0x1` (read paired).

Reply:
aaaah that explains why I was so confused. Thank you!